### PR TITLE
feat (client): make converter API public

### PR DESCRIPTION
**Pull request by kevin-dp**
This PR extends the converter API with `encodeRow(s)`/`decodeRow(s)` methods and exports the API.
This API is a nice to have when dealing with raw queries and having to correctly serialise/deserialise rows to/from the database.

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,6 +1,6 @@
 elixir 1.17.0-otp-27
 erlang 27.0
-nodejs 20.2.0
+nodejs 21.4.0
```
Comment (msfstef) [inline .tool-versions:3]: might need to update the CI workflows to use this version as well in the test matrix, build, etc

File: clients/typescript/src/client/conversions/converter.ts [hunk 2]
```diff
@@ -26,3 +64,34 @@ export interface Converter {
 export function isDataObject(v: unknown): boolean {
   return v instanceof Date || typeof v === 'bigint' || ArrayBuffer.isView(v)
 }
+
+export function mapRow<
+  T extends Record<string, unknown> = Record<string, unknown>
```
Comment (msfstef) [inline clients/typescript/src/client/conversions/converter.ts:69]: nit: might be worth extracting `Record<string, unknown>` to a type alias like `RowRecord` or something as it's so common

File: clients/typescript/src/client/conversions/converter.ts [hunk 3]
```diff
@@ -26,3 +64,34 @@ export interface Converter {
...
+>(
+  row: Record<string, unknown>,
+  tableSchema: Pick<AnyTableSchema, 'fields'>,
+  f: (v: any, pgType: PgType) => any
```
Comment (msfstef) [inline clients/typescript/src/client/conversions/converter.ts:73]: nit: more explicit name like `convert`? `f` feels a little bit _too_ terse haha

File: clients/typescript/src/client/conversions/converter.ts [hunk 4]
```diff
@@ -26,3 +64,34 @@ export interface Converter {
...
+
+  return mappedRow
+}
+
+export function mapRows<
+  T extends Record<string, unknown> = Record<string, unknown>
+>(
+  rows: Array<Record<string, unknown>>,
+  tableSchema: Pick<AnyTableSchema, 'fields'>,
+  f: (v: any, pgType: PgType) => any
```
Comment (msfstef) [inline clients/typescript/src/client/conversions/converter.ts:94]: nit: same as above

File: clients/typescript/src/client/conversions/postgres.ts [hunk 5]
```diff
@@ -11,7 +12,7 @@ import { PgBasicType, PgDateType, PgType } from './types'
  * Currently, no conversions are needed for the data types we support.
  */
 
-function toPostgres(v: any, pgType: PgType): any {
+export function toPostgres(v: any, pgType: PgType): any {
```
Comment (msfstef) [inline clients/typescript/src/client/conversions/postgres.ts:15]: nit: why are these exported, they are exported via the converter interface no? (same for SQLite)

File: clients/typescript/src/client/model/table.ts [hunk 6]
```diff
@@ -221,6 +225,10 @@ export class Table<
    * As such, one can compose these methods arbitrarily and then run them inside a single transaction.
    */
 
+  /**
```
Comment (msfstef) [inline clients/typescript/src/client/model/table.ts:228]: nit: might need more rebasing? haha

**Review by msfstef — APPROVED** [review]
Looks good, nice work! Left a few minor comments

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues and PRs. We really appreciate you taking the time to investigate and create this improvement!

**Pull request closed without merging**